bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/io/RefreshingOAuth2CredentialsInterceptorTest.java

```diff
@@ -96,26 +127,52 @@ public void testStaleAndExpired() throws IOException {
   }
 
   @Test
+  public void testRetries() throws Exception {
+    IOException ioException = new IOException("something bad happend");
+    Mockito.when(credentials.refreshAccessToken()).thenThrow(
+      ioException);
+    setTime(0);
+    underTest =
+        new RefreshingOAuth2CredentialsInterceptor(executorService, credentials, retryOptions);
+    underTest.sleeper = new Sleeper() {
+      @Override
+      public void sleep(long ms) throws InterruptedException {
+        setTime(RefreshingOAuth2CredentialsInterceptor.clock.currentTimeMillis() + ms);
+      }
+    };
+    HeaderCacheElement header = underTest.refreshCredentialsWithRetry(null);
+
+    Assert.assertNull(header.header);
+    Assert.assertSame(ioException, header.exception);
+
+    // Make sure that the system "slept" for more than the retryOption max millis. The Backoff logic
+    // adds some random variability to the exact elapsed time, so add in a bit of wiggle room.
+    long time = RefreshingOAuth2CredentialsInterceptor.clock.currentTimeMillis() / 1000;
```

Comment:
It is not clear to me why we need to divide current time with 1000 here.
T1 = start time
T2 = RefreshingOAuth2CredentialsInterceptor.clock.currentTimeMillis() // current time

T2 - T1 should be >= max duration if retry exhausted, no?